File: activityfrag/src/main/java/com/activityutil/activity/broadcast/LanguageBroadCastReceiver.java
package com.activityutil.activity.broadcast;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;

import com.activityutil.activity.activity.BaseAppCompatActivity;

/**
 * Created by clickapps on 24/2/16.
 * <p class="note">Language change Broadcast is included in this {@link LanguageBroadCastReceiver}.
 * Whenever language changes this broadcast hits  * </p>
 */
public class LanguageBroadCastReceiver extends BroadcastReceiver {
    private Activity mActivity;

    /**
     * Instantiates a new Language broad cast receiver.
     *
     * @param activity the activity
     */
    public LanguageBroadCastReceiver(@NonNull Activity activity) {
        this.mActivity = activity;
    }


    @Override
    public void onReceive(Context context, Intent intent) {
        if (mActivity instanceof BaseAppCompatActivity) {
            BaseAppCompatActivity activity = (BaseAppCompatActivity) mActivity;
            activity.recreate();
        }
    }

}

File: activity/src/main/java/com/common/broadcast/LanguageBroadCastReceiver.kt
package com.common.broadcast

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent


/**
 * Created by clickapps on 24/2/16.
 *
 * Language change Broadcast is included in this [LanguageBroadCastReceiver].
 * Whenever language changes this broadcast hits  *
 */
class LanguageBroadCastReceiver
/**
 * Instantiates a new Language broad cast receiver.
 *
 * @param activity the activity
 */
(private val mActivity: Activity) : BroadcastReceiver() {


    override fun onReceive(context: Context, intent: Intent) {
        mActivity.recreate()
    }

}
